controllers/drplacementcontrol_controller.go

```diff
@@ -371,6 +371,16 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{RequeueAfter: time.Second * initialWaitTime}, nil
 	}
 
+	// We've passed all initial checks and setup. At this point, the DRPC status condition
+	// should be Available
+	reason := rmn.ReasonSuccess
+	if d.instance.Status.Phase != "" {
+		reason = string(d.instance.Status.Phase)
+	}
+
+	SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionAvailable,
```

Comment:
(question) Would it be better to unconditionally update Available condition [here](https://github.com/RamenDR/ramen/blob/1869de906ce4fad7d51d7a9f39b8f7f1dba3d695/controllers/drplacementcontrol.go#L174), rather than start with a success state and then expect rest of the code to update failed/progress conditions? This seems to be the case already for failover/relocate, only initial deployment avoids resetting the Available condition to what it thinks it should be.

The rest of the code seems to handle changing the initial condition if set as above, but thinking about being safer and setting this once we know it is a success instead.

Reply:
You are right. I'll restructure it.